File: php/src/Resource/TaskContexts.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Resource;

use Fivexer\SDK\Fivexer;
use Fivexer\SDK\Model\SetTaskContext;
use Fivexer\SDK\Model\TaskContext;

/** The rich data stored against a task. Writes replace the stored context wholesale. */
final class TaskContexts
{
    public function __construct(private readonly Fivexer $client)
    {
    }

    public function get(string $taskId): TaskContext
    {
        return TaskContext::fromArray(
            $this->client->request('GET', '/tasks/' . \rawurlencode($taskId) . '/context') ?? []
        );
    }

    public function set(string $taskId, SetTaskContext $input): TaskContext
    {
        return TaskContext::fromArray(
            $this->client->request('PUT', '/tasks/' . \rawurlencode($taskId) . '/context', $input->toArray()) ?? []
        );
    }

    public function clear(string $taskId): void
    {
        $this->client->request('DELETE', '/tasks/' . \rawurlencode($taskId) . '/context');
    }
}

File: java/src/main/java/io/fivexer/sdk/TaskContexts.java
package io.fivexer.sdk;

import io.fivexer.sdk.internal.Json;
import io.fivexer.sdk.model.SetTaskContext;
import io.fivexer.sdk.model.TaskContext;

/** The rich data stored against a task. Writes replace the stored context wholesale. */
public final class TaskContexts {

    private final Fivexer client;

    TaskContexts(Fivexer client) {
        this.client = client;
    }

    public TaskContext get(String taskId) {
        return client.request("GET", "/tasks/" + Json.enc(taskId) + "/context", null, null,
                TaskContext.class, false);
    }

    public TaskContext set(String taskId, SetTaskContext input) {
        return client.request("PUT", "/tasks/" + Json.enc(taskId) + "/context", input.toJson(), null,
                TaskContext.class, false);
    }

    public void clear(String taskId) {
        client.request("DELETE", "/tasks/" + Json.enc(taskId) + "/context", null, null, Void.class, true);
    }
}
